conbench/app/_plots.py

```diff
@@ -258,8 +293,102 @@ def _inspect_for_multisample(items) -> tuple[bool, Optional[int]]:
 
 
-def time_series_plot(history, benchmark, run, height=380, width=1100):
+def gen_js_callback_tap_detect_unselect(source: bokeh.models.ColumnDataSource):
+    """
+    Dynamically manipulate Bootstrap panel conbench-histplot-run-details`.
+
+    Hide if empty space in plot was clicked.
+
+    Warning: requires manual testing, not covered by tests/CI.
+    """
+    return bokeh.models.CustomJS(
+        # Note(JP): When not using `args`, I thought, one can use
+        # `cb_data.source.selected.indices` to detect the situation where
+        # nothing was selected. However, for such events `cb_data` is an empty
+        # object. When using `args={"s1": source_mean_over_time}` then
+        # `s1.selected.indices` seems to always be available no matter where
+        # one clicks, and in case no glyph was clicked the array is zero
+        # length.
+        args={"s1": source},
+        code="""
+        // console.log("cb_data:", cb_data);
+        // console.log("cb_obj:", cb_obj);
+        // console.log("s1.selected.indices: ", s1.selected.indices);
+
+        if (s1.selected.indices.length == 0){
+            console.log("nothing selected, remove detail");
+            // Make the panel invisible.
+            const e = document.getElementsByClassName("conbench-histplot-run-details")[0];
+            e.style.display = 'none'
+        }
+    """,
+    )
+
 
+def gen_js_callback_click_on_glyph_show_run_details(repo_string):
+    """
+    Dynamically manipulate Bootstrap panel conbench-histplot-run-details`.
+
+    Show run details if a corresponding glyph was clicked in the plot.
+
+    Warning: requires manual testing, not covered by tests/CI.
+    """
+    return bokeh.models.CustomJS(
+        code=f"""
+
+        const i = cb_data.source.selected.indices[0];
+        const run_report_relurl = cb_data.source.data['relative_benchmark_urls'][i];
+        const run_date_string = cb_data.source.data['date_strings'][i];
+        // Remove first character, the hashtag
+        const run_commit_hash_short = cb_data.source.data['commit_hashes_short'][i].substring(1);
+        const run_commit_msg_pfx = cb_data.source.data['commit_messages'][i];
+        // This is a stringified version of the value that determines the y
+        // plot position of this glyph.
+        const run_result_value_with_unit = cb_data.source.data['values_with_unit'][i];
+
+        var run_samples_with_units = undefined;
+
+        if (cb_data.source.data['multisample_strings_with_unit'] !== undefined) {{
+            run_samples_with_units = cb_data.source.data['multisample_strings_with_unit'][i];
+        }}
+
+        // JavaScript code generated in a Python f string -- templating hell, yeah! :)
+        // I don't know if repo string is always a URL, if it's always
+        // pointing to GitHub. But if it does, we can do some UX sugar.
+
+        var commit_repo_string =  run_commit_hash_short + ' in {repo_string}';
+
+        const repo = '{repo_string}';
```

Comment:
That's the only place where it's dynamic. Pragmatic solution, I think it's actually fine for now.